### ISSUE TITLE
`box_pointers` lint warning in generated protobuf code

### ISSUE THREAD

**Issue by nayuta723** [checkpoint-restore/rust-criu#28]
## Problem

When building with recent Rust versions (1.91+), the following warning is emitted:

```
warning: lint `box_pointers` has been removed: it does not detect other kinds of allocations, and existed only for historical reasons
  --> src/rust_criu_protobuf/rpc.rs:12:10
   |
12 | #![allow(box_pointers)]
   |          ^^^^^^^^^^^^
   |
   = note: `#[warn(renamed_and_removed_lints)]` on by default
```

This is because `rust-protobuf` 3.2.0 generates code with `#![allow(box_pointers)]`, but this lint has been removed from Rust.

### Option 1: Add lint rule workaround in `build.rs`

Add post-processing in `build.rs` to insert `#![allow(renamed_and_removed_lints)]` before `#![allow(box_pointers)]` in the generated code.

### Option 2: Update protobuf to newer version

Upgrade `protobuf` and `protobuf-codegen` to a newer version that doesn't generate the deprecated lint.
ref: https://github.com/checkpoint-restore/rust-criu/pull/27

## Question
@adrianreber 
Which approach should we take to resolve this warning?
As far as I can see, the generated code contains no breaking changes. Therefore, I’d like to go with Option 2 (updating the protobuf). Does that sound okay to you?

**Comment by adrianreber** [comment]
Should be fixed. I took your commit.
